doc/architectural/folder-walkthrough.md

```diff
@@ -82,3 +82,110 @@ develop regression tests.
 The `unit/` directory contains the unit test suites. See
 \ref compilation-and-development for information on how to run and
 develop unit tests.
+
+## Directory dependencies ##
+
+\dot
+digraph directory_dependencies {
+  node [style = filled, color = white];
+
+  subgraph cluster_executables {
+    label = "Executables";
+    style = filled;
+    color = lightgrey;
+    node [style = filled, color = white];
+    CBMC [URL = "\ref cbmc"];
+    goto_cc [label = "goto-cc", URL = "\ref goto-cc"];
+    goto_analyzer [label = "goto-analyzer", URL = "\ref goto-analyzer"];
+    goto_instrument [label = "goto-instrument", URL = "\ref goto-instrument"];
+    goto_diff [label = "goto-diff", URL = "\ref goto-diff"];
+    janalyzer [URL = "\ref janalyzer"];
+    jdiff [URL = "\ref jdiff"];
+    JBMC [URL = "\ref jbmc"];
+    CBMC -> goto_instrument;
+    janalyzer -> goto_analyzer;
+    jdiff -> goto_diff;
+    JBMC -> CBMC;
+  }
+
+  subgraph cluster_symbolic_execution {
+    label = "Symbolic Execution";
+    style = filled;
+    color = lightgrey;
+    node [style = filled, color = white];
+    goto_symex [label = "goto-symex", URL = "\ref goto-symex"];
+  }
+
+  subgraph cluster_abstract_interpretation {
+    label = "Abstract Interpretation";
+    style = filled;
+    color = lightgrey;
+    node [style = filled, color = white];
+    pointer_analysis [label = "pointer-analysis", URL = "\ref pointer-analysis"];
+    analyses [URL = "\ref analyses"];
+    pointer_analysis -> analyses;
+  }
+
+  subgraph cluster_goto_programs {
+    label = "Goto Programs";
+    style = filled;
+    color = lightgrey;
+    goto_programs [label = "goto-programs", URL = "\ref goto-programs"];
+    linking [URL = "\ref linking"];
+    linking -> goto_programs;
+  }
+
+  subgraph cluster_solvers {
+    label = "Solvers"
+    style = filled;
+    color = lightgrey;
+    solvers [URL = "\ref solvers"];
+  }
+
+  subgraph cluster_languages {
+    label = "Languages";
+    style = filled;
+    color = lightgrey;
+    ansi_c [label = "ansi-c", URL = "\ref ansi-c"];
+    langapi [URL = "\ref langapi"];
+    cpp [URL = "\ref cpp"];
+    jsil [URL = "\ref jsil"];
+    java_bytecode [URL = "\ref java_bytecode"];  
+    ansi_c -> langapi;
+    cpp -> langapi;
+    jsil -> langapi;
+    java_bytecode -> langapi;
+  }
+
+  subgraph cluster_utilities {
+    label = "Utilities";
+    style = filled;
+    color = lightgrey;
+    big_int [label = "big-int", URL = "\ref big-int"];
+    miniz [URL = "\ref miniz"];
+    util [URL = "\ref util"];
+    nonstd [URL = "\ref nonstd"];
+    json [URL = "\ref json"];
+    xmllang [URL = "\ref xmllang"];
+    util -> big_int;
+    util -> nonstd;
+    util -> json;
+    util -> xmllang;
```

Comment:
Is that actually the case?

Reply:
No, it should be json -> util, xmllang -> util and then goto-programs uses json and xmllang.